open-hackathon-adminUI/src/app/admin/admin_mgr.py

```diff
@@ -138,21 +119,41 @@ def get_admin_info(self, admin):
             "last_login_time": str(admin.last_login_time)
         }
 
-#    def get_admin_detail_info(self, admin, **kwargs):
-#        detail = self.get_admin_info(admin)
-#
-#        experiments = admin.experiments.filter_by(status=ExprStatus.Running).all()
-#        detail["experiments"] = []
-#        map(lambda e: detail["experiments"].append({
-#            "id": e.id,
-#            "hackathon_id": e.hackathon_id
-#        }), experiments)
-#        detail["register_state"] = True
-#        hack = hack_manager.get_hackathon_by_name(kwargs['hackathon_name'])
-#        if hack is not None and safe_get_config('checkRegister', False) == True:
-#            detail["register_state"] = self.__validate_admin_registered(admin, hack)
-
-#        return detail
+    def get_hackid_from_adminid(self, admin_id):
+
+        # can not use backref in db models
+
+        # get emails from admin though admin.id in table admin_email
+        admin_emails = self.db.find_all_objects_by(AdminEmail, admin_id=admin_id)
+        emails = map(lambda x: x.email, admin_emails)
+
+        # get AdminUserHackathonRels from query withn filter by email
+        admin_user_hackathon_rels = self.db.find_all_objects(AdminUserHackathonRel,
+                                                             AdminUserHackathonRel.admin_email.in_(emails))
+        if admin_user_hackathon_rels is None:
```

Comment:
yeah~my fault……
an empty List is not None ，use len()==0 to check is right choise~

fixed~